### ISSUE TITLE
jruby -C uri:classloader:/ -e '`/bin/hostname`'

### ISSUE THREAD

**Issue by mkristian** [jruby/jruby#3111]
fails with
```
ava/lang/ProcessBuilder.java:1047:in `start': java.io.IOException: Cannot run program "/bin/hostname" (in directory "uri:classloader:"): error=2, No such file or directory
	from java/lang/reflect/Method.java:606:in `invoke'
	from file:/home/christian/projects/active/jruby/jruby17/lib/jruby.jar!/jruby/kernel/jruby/process_manager.rb:36:in ``'
	from file:/home/christian/projects/active/jruby/jruby17/lib/jruby.jar!/jruby/kernel/jruby/process_manager.rb:54:in ``'
	from -e:1:in `(root)'
	from ruby/-e:1:in `(root)'
Caused by:
UNIXProcess.java:-2:in `forkAndExec': java.io.IOException: error=2, No such file or directory
	from UNIXProcess.java:186:in `<init>'
	from ProcessImpl.java:130:in `start'
	from ProcessBuilder.java:1028:in `start'
	from NativeMethodAccessorImpl.java:-2:in `invoke0'
	from NativeMethodAccessorImpl.java:57:in `invoke'
	from DelegatingMethodAccessorImpl.java:43:in `invoke'
	from Method.java:606:in `invoke'
	from JavaMethod.java:441:in `invokeDirectWithExceptionHandling'
	from JavaMethod.java:305:in `invokeDirect'
	from InstanceMethodInvoker.java:35:in `call'
	from CachingCallSite.java:306:in `cacheAndCall'
	from CachingCallSite.java:136:in `call'
	from CallNoArgNode.java:60:in `interpret'
	from LocalAsgnNode.java:123:in `interpret'
	from NewlineNode.java:105:in `interpret'
	from BlockNode.java:71:in `interpret'
	from ASTInterpreter.java:74:in `INTERPRET_METHOD'
	from InterpretedMethod.java:182:in `call'
	from DefaultMethod.java:203:in `call'
	from CachingCallSite.java:326:in `cacheAndCall'
	from CachingCallSite.java:170:in `call'
	from CallOneArgNode.java:57:in `interpret'
	from NewlineNode.java:105:in `interpret'
	from ASTInterpreter.java:74:in `INTERPRET_METHOD'
	from InterpretedMethod.java:182:in `call'
	from DefaultMethod.java:203:in `call'
	from CachingCallSite.java:326:in `cacheAndCall'
	from CachingCallSite.java:170:in `call'
	from -e:1:in `__file__'
	from -e:-1:in `load'
	from Ruby.java:867:in `runScript'
	from Ruby.java:860:in `runScript'
	from Ruby.java:729:in `runNormally'
	from Ruby.java:578:in `runFromMain'
	from Main.java:395:in `doRunFromMain'
	from Main.java:290:in `internalRun'
	from Main.java:217:in `run'
	from Main.java:197:in `main'
```

**Comment by rtyler** [comment]
Great catch @blaines!

**Comment by blaines** [comment]
Thanks :thumbsup:

**Comment by mkristian** [comment]
fixed by 1fd27e60c6c036585d35c74f9735ae7a08ed746a

**Comment by rtyler** [comment]
@enebo this change looks like it was included in the 1.7.21 release but it wasn't in the right milestone, can you or @mkristian confirm and close?

**Comment by mkristian** [comment]
it is part of 1.7.21
on master though I do see 
```
sh: 1: cd: can't cd to uri:classloader://
```
